index.js

```diff
@@ -96,43 +96,79 @@ original error: ${e}`);
   }
 }
 
-async function exportPodcasts(podcastsDBData) {
+
+async function buildPodcastDict(fileName, cacheFilesPath, podcastsDBData) {
+  const uuid = fileName.replace(".mp3", "");
+  const dbMeta = podcastsDBData.find((m) => m.zuuid === uuid);
+  const path = `${cacheFilesPath}/${fileName}`;
+  const exportBase = dbMeta?.zcleanedtitle // 1. from apple podcast database
+        ?? (await getMP3MetaTitle(path)) // 2. from mp3 meta data
+        ?? uuid; // 3. fallback to unreadable uuid
+  const podcastName = dbMeta?.zpodcast.replaceAll('/', '_');
+  const exportFileName = sanitize(exportBase.substr(0, fileNameMaxLength));
+  const date = dbMeta?.date
+
+  const ret = {
+    podcastName,
+    date,
+    fileName,
+    path,
+    uuid,
+    exportFileName: `${exportFileName}.mp3`
+  };
+  return ret;
+}
+
+
+function filterPodcasts(podcasts, filepatterns = []) {
+  if (filepatterns.length == 0) {
+    return podcasts;
+  }
+
+  function matchesAny(s) {
+    return filepatterns.some((p) => { return s.indexOf(p) != -1 })
+  }
+
+  return podcasts.filter((p) => {
+    return matchesAny(p.exportFileName) || matchesAny(p.podcastName);
+  });
+}
+
+
+async function exportPodcasts(podcastsDBData, filepatterns = []) {
   const cacheFilesPath = await getPodcastsCacheFilesPath();
   const podcastMP3Files = await getPodcastsCacheMP3Files(cacheFilesPath);
-  const filesWithDBData = podcastMP3Files.map((fileName) => {
-    const uuid = fileName.replace(".mp3", "");
-    const dbMeta = podcastsDBData.find((m) => m.zuuid === uuid);
-    return {
-      fileName,
-      uuid,
-      path: `${cacheFilesPath}/${fileName}`,
-      dbMeta
-    };
-  });
+  const podcasts = await Promise.all(podcastMP3Files.map((fileName) => {
+    return buildPodcastDict(fileName, cacheFilesPath, podcastsDBData);
```

Comment:
to be honest I think `buildPodcastDict` is not super descriptive, even though it's my code I spent some time on understanding what is going on here, maybe something like `mergeFilesWithDBMetaData`

Reply:
Naming is tough!  Happy to change.

Reply:
Changed.